client/zanata-client-commands/src/main/java/org/zanata/client/commands/PutUserCommand.java

```diff
@@ -21,22 +21,46 @@ public PutUserCommand(PutUserOptions opts) {
     }
 
     public void run() throws Exception {
-        Account account = new Account();
-        account.setEmail(getOpts().getUserEmail());
-        account.setName(getOpts().getUserName());
-        account.setUsername(getOpts().getUserUsername());
-        account.setPasswordHash(getOpts().getUserPasswordHash());
-        account.setApiKey(getOpts().getUserKey());
+
+        String userUsername = getOpts().getUserUsername();
+        String passwordHash = getOpts().getUserPasswordHash();
+        String userEmail = getOpts().getUserEmail();
+
+        Account account = getClientFactory().getAccountClient().get(userUsername);
+
+        if (account == null) {
+            log.info("Creating new account {}", userUsername);
+            account = new Account();
+            if (userEmail == null || userUsername == null) {
+                throw new RuntimeException("New user's name and email must be specified");
+            } else if (passwordHash == null || StringUtils.isBlank(passwordHash)) {
+                log.warn("Unusable passwordHash set, user will need to use the Forgot Password feature");
+                account.setPasswordHash("NULL");
+            }
+            account.setUsername(userUsername);
+        } else {
+            log.info("Updating account {}", account.getUsername());
+        }
+
+        account.setEmail(Optional.ofNullable(userEmail).orElse(account.getEmail()));
+        account.setName(Optional.ofNullable(getOpts().getUserName()).orElse(account.getName()));
+        account.setApiKey(Optional.ofNullable(getOpts().getUserKey()).orElse(account.getApiKey()));
+        account.setPasswordHash(Optional.ofNullable(passwordHash).orElse(account.getPasswordHash()));
         account.setEnabled(!getOpts().isUserDisabled());
+
         Splitter splitter = Splitter.on(",").trimResults().omitEmptyStrings();
-        account.setRoles(Sets.newHashSet(splitter.split(getOpts().getUserRoles())));
-        account.setTribes(
-                Sets.newHashSet(splitter.split(getOpts().getUserLangs())));
+        if (getOpts().getUserRoles() != null) {
+            account.setRoles(Sets.newHashSet(splitter.split(getOpts().getUserRoles())));
+        }
+        if (getOpts().getUserLangs() != null) {
+            account.setTribes(
+                    Sets.newHashSet(splitter.split(getOpts().getUserLangs())));
+        }
 
         log.debug("{}", account);
```

Comment:
Regarding our discussion earlier, I figured that the problem lies with selection of admins rather than the client's display of passwordHash. A user could add the line back themselves and get the same result, provided they were an admin.